web/satellite/src/components/common/SearchComponent.vue

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+<template>
+    <input
+        @mouseenter="onMouseEnter"
+        @mouseleave="onMouseLeave"
+        @input="onInput"
+        v-model="searchQuery"
+        :placeholder="`Search ${placeHolder}`"
+        :style="customWidth"
+        type="text">
+</template>
+
+<script lang="ts">
+    import { Component, Prop, Vue } from 'vue-property-decorator';
+
+    declare type searchCallback = (search: string) => Promise<void>;
+
+    @Component
+    export default class SearchComponent extends Vue {
+        @Prop({default: ''})
```

Comment:
Is line 24 not enough for declaring the `default` prop?

Reply:
i'm not sure if i understand you right

there is '@Prop' decorator for each prop we have. Each prop has it's own default value so default value should be declared in every '@Prop' decorator.

here 24 line default value stands for 25 line prop and 22 line default value stands for 23 line prop.

Did you mean something else?

Reply:
A-ha, I see. You got it just right! Thanks for clarifying for me